src/openforms/products/static_variables/static_variables.py

```diff
@@ -0,0 +1,29 @@
+from typing import TYPE_CHECKING, Optional
+
+from django.utils.translation import gettext_lazy as _
+
+from openforms.variables.base import BaseStaticVariable
+from openforms.variables.constants import FormVariableDataTypes
+from openforms.variables.registry import register_static_variable
+
+if TYPE_CHECKING:  # pragma: nocover
+    from openforms.submissions.models import Submission
+
+
+@register_static_variable("product")
+class Product(BaseStaticVariable):
+    name = _("Product")
+    data_type = FormVariableDataTypes.object
+
+    def get_initial_value(self, submission: Optional["Submission"]) -> Optional[dict]:
```

Comment:
Fyi, since python 3.10 there is an equivalent way of writing `Optional["Submission"]` which is `"Submission" | None` (https://peps.python.org/pep-0604/). I think we tend to use the new syntax more.

Reply:
I get a `TypeError` when using `|` with a string with `None` (`Submission | None` would work if `Submission` was defined already but raises a `NameError` otherwise)